### ISSUE TITLE
Monitoring timeout while setup IP bwlim

### ISSUE THREAD

**Issue by JD-ISWT** [haproxy/haproxy#2321]
### Detailed Description of the Problem

I try to setup shared bandwidth limitation. 

My objective is to setup a shared limit per IP. When enabled, the limit is well applied but I got timeouts on response time from monitoring.
Monitoring is just a simple HTTP GET

![image](https://github.com/haproxy/haproxy/assets/57483715/e9fc59a4-3584-42af-80c7-bb43468879c7)
Limit disabled à 11:05

### Expected Behavior

BW limit applied without impact on response time

### Steps to Reproduce the Behavior

1. Setup shared bandwith limit using filter `bwlim-out` 
2. Enable the limit with `http-response set-bandwith-limit`

### What is your configuration?

```haproxy
frontend front_all_https
	bind :443 tfo ssl crt /etc/ssl/private/haproxy/<certname1> crt /etc/ssl/private/haproxy/<certname2> alpn h2,http/1.1
	mode http
	option http-keep-alive
	option forwardfor

	# --------------------------------------------
	# Path ACLs
	# --------------------------------------------
	acl path_admin path_beg /<path_admin1>
	acl path_admin path_beg /<path_admin2>

	acl path_public path_beg /<publicpath1>

	acl path_downloads path_beg /<downloadpath1>
	acl path_downloads path_beg /<downloadpath2>
	acl path_downloads path_beg /<downloadpath3>

	acl path_status path /status.php



	# Note: disabled for now, as it can break the streaming of videos in mobile apps
	#http-request tarpit deny_status 403 if path_direct_link

	# --------------------------------------------
	# Rate limiting (max 1000 req per 10s per IP)
	# --------------------------------------------
	http-request track-sc0 src table st_src_http_rate
	http-request deny deny_status 429 if { src,table_http_req_rate(st_src_http_rate) gt 1000 } !is_src_whitelist

	# --------------------------------------------
	# Traffic Shaping per stream
	# --------------------------------------------
	filter bwlim-out global_download default-limit 10m default-period 1s # 10MB/s per stream
	http-request set-bandwidth-limit global_download limit 10M period 1s if path_downloads   # 10MB/s per stream
	http-request set-bandwidth-limit global_download limit 5M period 1s if path_public # 5MB/s  per stream

	# --------------------------------------------
	# Traffic Shaping per source IP
	# --------------------------------------------
	filter bwlim-out limit_by_src limit 50M key src table st_src_bw # 50MB/s per IP
	http-response set-bandwidth-limit limit_by_src
[...]

backend st_src_bw
	stick-table type ipv6 size 1m expire 3600s store bytes_out_rate(1s)

backend st_src_http_rate
	stick-table type ipv6 size 100k expire 30s store http_req_rate(10s)
```

### Output of `haproxy -vv`

```plain
HAProxy version 2.8.3-1~bpo11+1 2023/09/08 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2028.
Known bugs: http://www.haproxy.org/bugs/bugs-2.8.3.html
Running on: Linux 5.10.0-26-cloud-amd64 #1 SMP Debian 5.10.197-1 (2023-09-29) x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wundef -Wdeclaration-after-statement -Wfatal-errors -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wno-string-plus-int -Wno-atomic-alignment
  OPTIONS = USE_OPENSSL=1 USE_LUA=1 USE_SLZ=1 USE_SYSTEMD=1 USE_PROMEX=1 USE_PCRE2=1 USE_PCRE2_JIT=1
  DEBUG   = -DDEBUG_STRICT -DDEBUG_MEMORY_POOLS

Feature list : -51DEGREES +ACCEPT4 +BACKTRACE -CLOSEFROM +CPU_AFFINITY +CRYPT_H -DEVICEATLAS +DL -ENGINE +EPOLL -EVPORTS +GETADDRINFO -KQUEUE -LIBATOMIC +LIBCRYPT +LINUX_CAP +LINUX_SPLICE +LINUX_TPROXY +LUA +MATH -MEMORY_PROFILING +NETFILTER +NS -OBSOLETE_LINKER +OPENSSL -OPENSSL_WOLFSSL -OT -PCRE +PCRE2 +PCRE2_JIT -PCRE_JIT +POLL +PRCTL -PROCCTL +PROMEX -PTHREAD_EMULATION -QUIC +RT +SHM_OPEN +SLZ +SSL -STATIC_PCRE -STATIC_PCRE2 +SYSTEMD +TFO +THREAD +THREAD_DUMP +TPROXY -WURFL -ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_TGROUPS=16, MAX_THREADS=256, default=8).
Built with OpenSSL version : OpenSSL 1.1.1n  15 Mar 2022
Running on OpenSSL version : OpenSSL 1.1.1w  11 Sep 2023 (VERSIONS DIFFER!)
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.3
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.36 2020-12-04
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 10.2.1 20210110

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
         h2 : mode=HTTP  side=FE|BE  mux=H2    flags=HTX|HOL_RISK|NO_UPG
       fcgi : mode=HTTP  side=BE     mux=FCGI  flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=HTTP  side=FE|BE  mux=H1    flags=HTX
         h1 : mode=HTTP  side=FE|BE  mux=H1    flags=HTX|NO_UPG
  <default> : mode=TCP   side=FE|BE  mux=PASS  flags=
       none : mode=TCP   side=FE|BE  mux=PASS  flags=NO_UPG

Available services : prometheus-exporter
Available filters :
	[BWLIM] bwlim-in
	[BWLIM] bwlim-out
	[CACHE] cache
	[COMP] compression
	[FCGI] fcgi-app
	[SPOE] spoe
	[TRACE] trace
```

### Additional Information

```
 echo "show table st_src_bw" | sudo -u haproxy socat stdio unix-connect:/var/run/haproxy/admin.sock 
# table: st_src_bw, type: ipv6, size:1048576, used:1
0x7fe2f00877d0: key=::ffff:<redactedIP1> use=0 exp=3587424 shard=0 bytes_out_rate(1000)=0

```

**Comment by capflam** [comment]
Could you share your haproxy logs and configuration of your defaults section please ?

**Comment by JD-ISWT** [comment]
Here is the defaults section

```
# ---------------------------------------------------------------------
# DEFAULTS
# ---------------------------------------------------------------------
defaults
	log global
	mode http
	option httplog
	option dontlognull
	option dontlog-normal
	option redispatch
	timeout client 6m
	timeout connect 3s
	timeout server 6m
	timeout http-request 30s
	timeout http-keep-alive 5s
	timeout queue 5s
	timeout tunnel 2m
	timeout client-fin 1s
	timeout server-fin 1s
	errorfile 400 /etc/haproxy/errors/400.http
	errorfile 403 /etc/haproxy/errors/403.http
	errorfile 408 /etc/haproxy/errors/408.http
	errorfile 500 /etc/haproxy/errors/500.http
	errorfile 502 /etc/haproxy/errors/502.http
	errorfile 503 /etc/haproxy/errors/503.http
	errorfile 504 /etc/haproxy/errors/504.http
```

There is nothing relevant in the logs

**Comment by JD-ISWT** [comment]
I enabled more log and see many `SL` as termination state, but don't know how to construe this information
```
Nov  6 14:19:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:9065 [06/Nov/2023:14:19:37.076] front_all_https~ <app_backend>/web4 0/0/1/101/1103 200 1511 - - SL-- 20/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 14:23:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:7821 [06/Nov/2023:14:23:37.040] front_all_https~ <app_backend>/web3 0/0/2/72/1076 200 1509 - - SL-- 23/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 14:25:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:65353 [06/Nov/2023:14:25:37.149] front_all_https~ <app_backend>/web3 0/0/1/75/1077 200 1503 - - SL-- 22/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 14:31:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:9704 [06/Nov/2023:14:31:37.215] front_all_https~ <app_backend>/web4 0/0/1/76/1077 200 1513 - - SL-- 21/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 14:38:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:62060 [06/Nov/2023:14:38:37.096] front_all_https~ <app_backend>/web3 0/0/2/74/1077 200 1507 - - SL-- 23/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 14:45:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:18217 [06/Nov/2023:14:45:37.225] front_all_https~ <app_backend>/web4 0/0/2/77/1080 200 1505 - - SL-- 22/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 14:48:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:57597 [06/Nov/2023:14:48:37.134] front_all_https~ <app_backend>/web4 0/0/2/77/1080 200 1511 - - SL-- 21/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 14:58:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:36812 [06/Nov/2023:14:58:37.055] front_all_https~ <app_backend>/web3 0/0/1/73/1076 200 1511 - - SL-- 42/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:05:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:40227 [06/Nov/2023:15:05:37.161] front_all_https~ <app_backend>/web3 0/0/2/73/1076 200 1507 - - SL-- 24/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:06:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:23908 [06/Nov/2023:15:06:37.239] front_all_https~ <app_backend>/web4 0/0/2/77/1079 200 1511 - - SL-- 21/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:09:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:3627 [06/Nov/2023:15:09:37.142] front_all_https~ <app_backend>/web4 0/0/2/75/1077 200 1505 - - SL-- 23/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:13:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:3236 [06/Nov/2023:15:13:37.100] front_all_https~ <app_backend>/web3 0/0/1/98/1102 200 1505 - - SL-- 23/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:16:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:2944 [06/Nov/2023:15:16:37.395] front_all_https~ <app_backend>/web4 0/0/2/75/1079 200 1511 - - SL-- 23/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:21:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:51074 [06/Nov/2023:15:21:37.022] front_all_https~ <app_backend>/web4 0/0/2/76/1079 200 1503 - - SL-- 21/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:24:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:29690 [06/Nov/2023:15:24:37.171] front_all_https~ <app_backend>/web3 0/0/2/75/1079 200 1507 - - SL-- 23/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:30:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:32317 [06/Nov/2023:15:30:37.382] front_all_https~ <app_backend>/web4 0/0/2/75/1078 200 1505 - - SL-- 33/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:32:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:55635 [06/Nov/2023:15:32:37.099] front_all_https~ <app_backend>/web4 0/0/2/77/1081 200 1511 - - SL-- 23/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:37:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:48779 [06/Nov/2023:15:37:37.540] front_all_https~ <app_backend>/web3 0/0/1/75/1079 200 1503 - - SL-- 25/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:41:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:40645 [06/Nov/2023:15:41:37.086] front_all_https~ <app_backend>/web4 0/0/1/78/1081 200 1503 - - SL-- 28/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:44:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:4799 [06/Nov/2023:15:44:37.242] front_all_https~ <app_backend>/web4 0/0/2/77/1079 200 1509 - - SL-- 35/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:47:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:22030 [06/Nov/2023:15:47:37.146] front_all_https~ <app_backend>/web4 0/0/1/95/1097 200 1509 - - SL-- 23/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:50:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:30772 [06/Nov/2023:15:50:37.057] front_all_https~ <app_backend>/web4 0/0/2/76/1079 200 1509 - - SL-- 31/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 15:53:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:7876 [06/Nov/2023:15:53:37.075] front_all_https~ <app_backend>/web3 0/0/1/74/1077 200 1505 - - SL-- 34/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:04:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:37933 [06/Nov/2023:16:04:37.080] front_all_https~ <app_backend>/web3 0/0/1/76/1078 200 1509 - - SL-- 22/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:07:38 haproxy1 haproxy[4092943]: <MONITORING_IP>:55040 [06/Nov/2023:16:07:37.501] front_all_https~ <app_backend>/web3 0/0/2/73/1076 200 1515 - - SL-- 21/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:13:37 haproxy1 haproxy[230258]: <MONITORING_IP>:48641 [06/Nov/2023:16:13:37.055] front_all_https~ <app_backend>/web3 0/0/2/73/75 200 1511 - - ---- 7/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:17:38 haproxy1 haproxy[230258]: <MONITORING_IP>:60331 [06/Nov/2023:16:17:37.249] front_all_https~ <app_backend>/web3 0/0/2/75/1079 200 1499 - - SL-- 9/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:18:37 haproxy1 haproxy[232373]: <MONITORING_IP>:39608 [06/Nov/2023:16:18:37.213] front_all_https~ <app_backend>/web3 0/0/2/72/74 200 1503 - - ---- 1/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:19:38 haproxy1 haproxy[232373]: <MONITORING_IP>:33762 [06/Nov/2023:16:19:37.103] front_all_https~ <app_backend>/web4 0/0/2/79/1082 200 1505 - - SL-- 5/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:21:37 haproxy1 haproxy[232373]: <MONITORING_IP>:18404 [06/Nov/2023:16:21:37.211] front_all_https~ <app_backend>/web3 0/0/1/72/73 200 1515 - - ---- 7/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:22:38 haproxy1 haproxy[232373]: <MONITORING_IP>:54261 [06/Nov/2023:16:22:37.148] front_all_https~ <app_backend>/web4 0/0/2/95/1099 200 1509 - - SL-- 7/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:23:37 haproxy1 haproxy[232373]: <MONITORING_IP>:34812 [06/Nov/2023:16:23:37.064] front_all_https~ <app_backend>/web3 0/0/1/74/75 200 1509 - - ---- 8/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:24:37 haproxy1 haproxy[232373]: <MONITORING_IP>:22476 [06/Nov/2023:16:24:37.116] front_all_https~ <app_backend>/web3 0/0/1/73/74 200 1511 - - ---- 11/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:26:37 haproxy1 haproxy[232373]: <MONITORING_IP>:35797 [06/Nov/2023:16:26:37.221] front_all_https~ <app_backend>/web3 0/0/1/73/74 200 1509 - - ---- 11/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:28:37 haproxy1 haproxy[232373]: <MONITORING_IP>:59869 [06/Nov/2023:16:28:37.086] front_all_https~ <app_backend>/web4 0/0/1/83/84 200 1501 - - ---- 9/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:31:37 haproxy1 haproxy[232373]: <MONITORING_IP>:48956 [06/Nov/2023:16:31:37.241] front_all_https~ <app_backend>/web3 0/0/2/88/90 200 1507 - - ---- 9/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:32:38 haproxy1 haproxy[232373]: <MONITORING_IP>:24643 [06/Nov/2023:16:32:37.045] front_all_https~ <app_backend>/web4 0/0/1/78/1081 200 1503 - - SL-- 9/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:34:37 haproxy1 haproxy[235812]: <MONITORING_IP>:4771 [06/Nov/2023:16:34:37.321] front_all_https~ <app_backend>/web3 0/0/1/74/75 200 1511 - - ---- 33/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:35:38 haproxy1 haproxy[235812]: <MONITORING_IP>:39757 [06/Nov/2023:16:35:37.212] front_all_https~ <app_backend>/web3 0/0/2/74/1078 200 1511 - - SL-- 36/1/0/0/0 0/0 "GET /status.php HTTP/1.0"
Nov  6 16:36:37 haproxy1 haproxy[235812]: <MONITORING_IP>:1606 [06/Nov/2023:16:36:37.013] front_all_https~ <app_backend>/web3 0/0/2/74/76 200 1505 - - ---- 22/1/0/0/0 0/0 "GET /status.php HTTP/1.0"

```

There are no more `SLs` once disabled

**Comment by capflam** [comment]
Thanks. I performed some tests and in fact I'm able to reproduce it. There is a bug in `freq_ctr_overshoot_period()` function. I'll try to fix is soon.

**Comment by capflam** [comment]
I push a fix. But I forgot to link the issue into the commit message. The issue should be fixed by 5705a6e3b7b310c87f21abaa710da66a0f38108a. It will be backported as usual. Thanks !

**Comment by JD-ISWT** [comment]
Thks

How long until it would be available in apt repositories?

**Comment by capflam** [comment]
Hard to say. We must backport the fix first and then emit a new release. The 2.8.4 will probably be released soon because the 2.8.3 is quite old now. But there is no ETA for now.
